src/methods.jl

```diff
@@ -101,9 +104,10 @@ end
 for fname in (:cor, :cov)
     @eval Statistics.$fname(A::AbDimArray{T,2}; dims=1, kwargs...) where T = begin
         newdata = Statistics.$fname(data(A); dims=dimnum(A, dims), kwargs...)
-        I = dims2indices(A, dims, 1)
-        newdims, newrefdims = slicedims(A, I)
-        rebuild(A, newdata, (newdims[1], newdims[1]), newrefdims)
+        removed_idx = dimnum(A, dims)
+        newrefdims = dims_alias(A)[removed_idx]
+        newdims = dims_alias(A)[abs(3 - removed_idx)]
+        rebuild(A, newdata, (newdims, newdims), (newrefdims,))
```

Comment:
This is ugly, but does the right thing. Any suggestions for a cleaner implementation?

Reply:
Maybe `getdims` instead of `dims_alias`. 

And  can you explain what `abs(3 - remove_idx)` is doing? Why do you need `abs` ?

Reply:
Oh, since I was thinking about some other code when I wrote it. I'll change that.

Reply:
Fixed. Also realized I could just escape `dims`.